app/forms/validators.py

```diff
@@ -22,12 +27,16 @@
 )
 email_regex = re.compile(r"^.+@([^.@][^@\s]+)$")
 
+OptionalMessage = Optional[Mapping[str, str]]
+NumType = Union[int, Decimal]
+PeriodType = Mapping[str, int]
+
 
 class NumberCheck:
-    def __init__(self, message=None):
+    def __init__(self, message: OptionalMessage = None):
         self.message = message or error_messages["INVALID_NUMBER"]
 
-    def __call__(self, form, field):
+    def __call__(self, form: FlaskForm, field: IntegerField) -> NoReturn:
```

Comment:
Yes, that throws error as circular reference hence had to use FlaskForm instead  
> Using `QuestionnaireForm` would be more appropriate, since we inherit from `FlaskForm` in that. Did you try questionnaire form?

Reply:
That is a common issue when typing, you can get it working by importing questionnaire form when typing checking only. So to use `QuestionnaireForm` you would need to import `TYPE_CHECKING` from `typing. Then you can reference it using quotes.

```py

from typing import TYPE_CHECKING
...

if TYPE_CHECKING:
    from app.forms.questionnaire_form import QuestionnaireForm
    
def __call__(self, form: "QuestionnaireForm", field: IntegerField) -> None:
	...
```